persistence/persistence-mybatis/src/main/resources/org/cbioportal/persistence/mybatis/SampleMapper.xml

```diff
@@ -20,4 +21,61 @@
             </include>
         </if>
     </sql>
+
+    <sql id="from">
+        FROM sample
+        INNER JOIN patient ON sample.PATIENT_ID = patient.INTERNAL_ID
+        INNER JOIN cancer_study ON patient.CANCER_STUDY_ID = cancer_study.CANCER_STUDY_ID
+    </sql>
+
+    <sql id="where">
+        <where>
+            <if test="sampleIds == null">
+                cancer_study.CANCER_STUDY_IDENTIFIER = #{studyIds[0]}
```

Comment:
We control this in the web layer, but if a 3rd party uses the persistence layer, it's not that clear. I'll think of a way to make this more clear for the users of persistence module.